service/src/test/java/com/epam/rft/atsy/service/impl/CandidateServiceImplTest.java

```diff
@@ -120,6 +132,46 @@ public void getCandidateShouldReturnProperDtoWhenIdExists() {
   }
 
   @Test(expected = IllegalArgumentException.class)
+  public void getCandidateByApplicationIDShouldThrowIllegalArgumentExceptionWhenApplicationIDIsNull() {
+    // When
+    candidateService.getCandidateByApplicationID(null);
+  }
+
+  @Test
+  public void getCandidateByApplicationIdShouldReturnNullWhenThereIsNoApplicationWithTheGivenId() {
+    // Given
+    given(applicationsRepository.findOne(APPLICATION_ID))
+        .willReturn(dummyApplicationEntity);
+    given(converterService.convert(dummyCandidateEntity, CandidateDTO.class))
```

Comment:
converterService will never be called using the `dummyCandidateEntity` in this test case, instead it will be invoked with a `null` parameter

I think the converter should not be called when the candidateEntity is null.